data-structures-stacks/max-value.js

```diff
@@ -1,7 +1,8 @@
 export default function maxValue(stack) {
-  const allValues = [];
+  let maxValue = -Infinity;
   while (stack.peek() !== undefined) {
-    allValues.push(stack.pop());
+    const top = stack.pop();
+    (maxValue < top) && (maxValue = top);
```

Comment:
This works, and it seems succinct, but I actually find it less readable than a simple `if`. It's too clever for my tastes!